frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,256 @@
+            isSuccessArtifacts &&
+            isSuccessArtifactTypes &&
+            artifacts &&
+            artifactTypes && (
+              <MetricsSwitcher
+                artifacts={artifacts}
+                artifactTypes={artifactTypes}
+                refresh={refresh}
+              ></MetricsSwitcher>
+            )}
+        </div>
+      </div>
+    </ErrorBoundary>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      id: artifact.getId(),
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {confidenceMetricsArtifacts &&
+        confidenceMetricsArtifacts.length > 0 &&
+        confidenceMetricsArtifacts.map(artifact => {
+          const confidenceMetrics = artifact.customProperties
+            .get('confidenceMetrics')
+            ?.getStructValue()
+            ?.toJavaScript();
+          const { error } = validateConfidenceMetrics((confidenceMetrics as any).list);
+
+          return (
+            <>
+              {error && (
+                <Banner
+                  message='Error in confidenceMetrics data format.'
+                  mode='error'
+                  additionalInfo={error}
+                />
+              )}
+              {!error && confidenceMetrics && (
+                <div key={'confidenceMetrics' + artifact.id}>
+                  <div className={padding(40, 'lrt')}>
+                    <h1>
+                      {'ROC Curve: ' + artifact.customProperties.get('name')?.getStringValue()}{' '}
+                      <IconWithTooltip
+                        Icon={HelpIcon}
+                        iconColor={color.weak}
+                        tooltip={ROC_CURVE_DEFINITION}
+                      ></IconWithTooltip>
+                    </h1>
+                  </div>
+                  <ROCCurve configs={buildRocCurveConfig((confidenceMetrics as any).list)} />
```

Comment:
Shall we move the error branch here, so that when error happens, we still have a title saying which artifact is in error?

Reply:
I am able to add title to error banner from the current location. Updated.